anta/tests/routing/bgp.py

```diff
@@ -1404,3 +1404,89 @@ def test(self) -> None:
             self.result.is_success()
         else:
             self.result.is_failure(f"The following BGP peers are not configured or have non-zero update error counters:\n{failures}")
+
+
+class VerifyBGPPeerRouteLimit(AntaTest):
+    """Verifies maximum routes and maximum routes warning limit for the provided BGP IPv4 peer(s).
+
+    Expected Results
+    ----------------
+    * Success: The test will pass if the BGP peer's maximum routes and maximum routes warning limit are equal to given limits.
+    * Failure: The test will fail if the BGP peer's maximum routes and maximum routes warning limit are not equal to given limits or
+    peer is not configured.
+
+    Examples
+    --------
+    ```yaml
+    anta.tests.routing:
+      bgp:
+        - VerifyBGPPeerRouteLimit:
+            bgp_peers:
+              - peer_address: 172.30.11.1
+                vrf: default
+                maximum_routes: 12000
+                warning_limit: 10000
+    ```
+    """
+
+    name = "VerifyBGPPeerRouteLimit"
+    description = "Verifies maximum routes and maximum routes warning limit for the provided BGP IPv4 peer(s)."
+    categories: ClassVar[list[str]] = ["bgp"]
+    commands: ClassVar[list[AntaCommand | AntaTemplate]] = [AntaTemplate(template="show bgp neighbors {peer} vrf {vrf}", revision=3)]
+
+    class Input(AntaTest.Input):
+        """Input model for the VerifyBGPPeerRouteLimit test."""
+
+        bgp_peers: list[BgpPeer]
+        """List of BGP peers"""
+
+        class BgpPeer(BaseModel):
+            """Model for a BGP peer."""
+
+            peer_address: IPv4Address
+            """IPv4 address of a BGP peer."""
+            vrf: str = "default"
+            """Optional VRF for BGP peer. If not provided, it defaults to `default`."""
+            maximum_routes: int
```

Comment:
can you check if we have limit for max route and warning route to restrict the input

Reply:
Updated input model docstring. Thanks!!

Reply:
Please use Pydantic Field instead to restrict the input. The docstring would be simplified after.